ValveResourceFormat/Resource/ResourceTypes/Material.cs

```diff
@@ -146,10 +146,7 @@ private KVObject GetInputSignatureObject()
                 return null;
             }
 
-            var extraStringData = (ExtraStringData)Resource.EditInfo.Structs[ResourceEditInfo.REDIStruct.ExtraStringData];
-            var inputSignatureString = extraStringData.List.Where(x => x.Name == "VSInputSignature").FirstOrDefault()?.Value;
-
-            if (inputSignatureString == null)
+            if (Resource.EditInfo.SearchableUserData.FirstOrDefault(x => x.Key == "VSInputSignature").Value is not string inputSignatureString)
```

Comment:
Is this not a null ref?

Reply:
Return type is `KeyValuePair<string, object>` so a valuetype. `Value` accessor may be null though, and that is correctly checked.

Reply:
as a value type, `default(KeyValuePair<string, object>>)` is not null